llama.cpp

```diff
@@ -4375,6 +4380,9 @@ static bool llm_load_tensors(
             case LLM_ARCH_MINICPM:
                 {
                     model.tok_embd = ml.create_tensor(ctx_input, tn(LLM_TENSOR_TOKEN_EMBD, "weight"), {n_embd, n_vocab});
+                    if (!hparams.tie_lm_head){
+                        model.output      = ml.create_tensor(ctx_output_split, tn(LLM_TENSOR_OUTPUT,      "weight"), {n_embd, n_vocab}, false);
+                    }
```

Comment:
We already handle tied tensors few lines below. Maybe you simply have to remove the `if (model.arch != LLM_ARCH_MINICPM){` check and this model would work?

Reply:
Done.

Reply:
Hello @ggerganov , we are currently encountering a problem when adapting our [new model](https://huggingface.co/openbmb/MiniCPM-2B-128k): the evaluation effect for models smaller than 4k is consistent with vllm, but the evaluation effect for models larger than 4k is not as good as vllm. What may be the reason for this? When evaluating, the example/server method is used, with the following startup and request parameters:
```
./server -m MiniCPM-2B-128k/ggml-model-f16.gguf --chat-template chatml --rope-freq-base 4129032.258 --host 0.0.0.0 -c 12000
```
request data:
```
   data = {"stream": False,
            "n_predict": max_token,
            "temperature": 0.3,
            "stop": ["<|im_end|>", "</s>"],
            "repeat_last_n": 256,
            "repeat_penalty": 1.0,
            "top_k": 40,
            "top_p": 0.5,
            "min_p": 0.05,
            "tfs_z": 1,
            "typical_p": 1,
            "presence_penalty": 0,
            "frequency_penalty": 0,
            "mirostat": 0,
            "mirostat_tau": 5,
            "mirostat_eta": 0.1,
            "grammar": "", "n_probs": 0, "min_keep": 0, "image_data": [], "cache_prompt": True,
            "api_key": "",
            "prompt": f"<|im_start|>user{prompt}<|im_end|><|im_start|>assistant\n"
            }
```

vllm param:
```
params_dict = {
    "n": 1,
    "best_of": None,
    "presence_penalty": 0.0, 
    "frequency_penalty": 0.0,
    "repetition_penalty": 1.0,
    "temperature": 0.3, 
    "top_p": 0.5, 
    "top_k": -1,
    "use_beam_search": False,
    "length_penalty": 1.0,
    "early_stopping": False,
    "stop": None,
    "stop_token_ids": None,
    "ignore_eos": False,
    "logprobs": None,
    "prompt_logprobs": None,
    "skip_special_tokens": False,
    "stop": ["<|im_end|>", "</s>"]
}
```

Reply:
The model currently uses [DynamicNTKScalingRotaryEmbedding](https://github.com/vllm-project/vllm/blob/d150e4f89f9da5b093511227449ad940a9a82a52/vllm/model_executor/layers/rotary_embedding.py#L200). How should I pass parameters?

Reply:
It can run up to 64k without NTK scaling.

Reply:
I think we don't support it. Anyway, isn't 64k context length enough?

https://github.com/ggerganov/llama.cpp/blob/148ec970b62c3c5ae0a8bfdaad2fc237aaae350d/convert_hf_to_gguf.py#L1638-L1650